satpy/readers/yaml_reader.py

```diff
@@ -407,6 +415,7 @@ def filename_items_for_filetype(filenames, filetype_info):
                     continue
                 matched_files.append(filename)
                 yield filename, filename_info
+            filenames -= set(matched_files)
```

Comment:
Why is `matched_files` first a list (line 407), then turned into a set here, rather than a set all the time?

Reply:
Good point. I think I was worried about if matched_files needed to be a certain order for the best performance. I think you're right though, I should just make everything a set instead of copying so many times.